beacon-chain/sync/payload_attestations.go

```diff
@@ -0,0 +1,106 @@
+	eth "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"go.opencensus.io/trace"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	errInvalidPayloadStatus   = errors.New("invalid PayloadStatus")
+	errInvalidBeaconBlockRoot = errors.New("invalid BeaconBlockRoot")
+	errInvalidValidatorIndex  = errors.New("invalid validator index")
+	errUnkownBeaconBlockRoot  = errors.New("unkonwn beacon block")
+	errNotInPTC               = errors.New("validator index not in Payload Timeliness Committee")
+)
+
+func (s *Service) validatePayloadAttestation(ctx context.Context, pid peer.ID, msg *pubsub.Message) (pubsub.ValidationResult, error) {
+	if pid == s.cfg.p2p.PeerID() {
+		return pubsub.ValidationAccept, nil
+	}
+	if s.cfg.initialSync.Syncing() {
+		return pubsub.ValidationIgnore, nil
+	}
+	ctx, span := trace.StartSpan(ctx, "sync.validatePayloadAttestation")
+	defer span.End()
+	if msg.Topic == nil {
+		return pubsub.ValidationReject, errInvalidTopic
+	}
+	m, err := s.decodePubsubMessage(msg)
+	if err != nil {
+		tracing.AnnotateError(span, err)
+		return pubsub.ValidationReject, err
+	}
+	att, ok := m.(*eth.PayloadAttestationMessage)
+	if !ok {
+		return pubsub.ValidationReject, errWrongMessage
+	}
+	if err := helpers.ValidateNilPayloadAttestationMessage(att); err != nil {
+		return pubsub.ValidationReject, err
+	}
+	if att.Data.Slot != s.cfg.clock.CurrentSlot() {
+		return pubsub.ValidationIgnore, nil
+	}
+	if att.Data.PayloadStatus >= primitives.PAYLOAD_INVALID_STATUS {
+		return pubsub.ValidationReject, errInvalidPayloadStatus
+	}
+	root := [32]byte(att.Data.BeaconBlockRoot)
+	if s.hasBadBlock(root) {
+		return pubsub.ValidationReject, errInvalidBeaconBlockRoot
+	}
+	if !s.cfg.chain.InForkchoice(root) {
+		return pubsub.ValidationIgnore, errUnkownBeaconBlockRoot
+	}
+	st, err := s.cfg.chain.HeadState(ctx)
+	if err != nil {
+		return pubsub.ValidationIgnore, err
+	}
+	if err := helpers.ValidatePayloadAttestationMessageSignature(ctx, st, att); err != nil {
+		return pubsub.ValidationReject, err
+	}
+	ptc, err := helpers.InPayloadTimelinessCommittee(ctx, st, att.Data.Slot, att.ValidatorIndex)
+	if err != nil {
+		return pubsub.ValidationIgnore, err
+	}
+	if !ptc {
+		return pubsub.ValidationReject, errNotInPTC
+	}
+	return pubsub.ValidationAccept, nil
+}
+
+func (s *Service) payloadAttestationSubscriber(ctx context.Context, msg proto.Message) error {
+	a, ok := msg.(*eth.PayloadAttestationMessage)
+	if !ok {
+		return errWrongMessage
+	}
+	if err := helpers.ValidateNilPayloadAttestationMessage(a); err != nil {
+		return err
+	}
+	root := [32]byte(a.Data.BeaconBlockRoot)
+	st, err := s.cfg.chain.HeadState(ctx)
+	if err != nil {
+		return err
+	}
+	ptc, err := helpers.GetPayloadTimelinessCommittee(ctx, st, a.Data.Slot)
+	if err != nil {
+		return err
+	}
+	idx := slice.Index(ptc, a.ValidatorIndex)
+	if idx == -1 {
+		return errInvalidValidatorIndex
+	}
+	if s.payloadAttestationCache.Seen(root, uint64(idx)) {
+		return nil
+	}
```

Comment:
This is in the p2p spec, just that it's a little harder to implement it cause we use the ptc index instead of the validator index to key the cache. I'll check on gossip as well.

Reply:
yeah the latest push already does this